include/AssetManager.h

```diff
@@ -27,8 +27,8 @@ class AssetManager
     void createProjectile(Vector2D pos, Vector2D velocity, int scale, int range, int speed, const char* texturePath, Entity* owner);
     void createPowerup(Vector2D pos, std::function<void (Entity*)> pickupFunc, std::string texturePath);
 
-    Vector2D calculateSpawnPosition();
-    PowerupType calculateType();
+    Vector2D calculateSpawnPosition(int width, int height);
+    template <typename T> T calculateType(int amount);
```

Comment:
`calculateType` does not seem like an intuitive name, please add random to its name.

Also does not seem to be used, maybe add `[[deprecated]]`